src/gt4py/gt_src_manager.py

```diff
@@ -28,29 +29,33 @@
 
 
 def install_gt_sources() -> bool:
-    is_ok = has_gt_sources()
-    if not is_ok:
-        GIT_BRANCH = _GRIDTOOLS_GIT_BRANCH
-        GIT_REPO = _GRIDTOOLS_GIT_REPO
+    if has_gt_sources():
+        return True
 
-        install_path = os.path.dirname(__file__)
-        target_path = os.path.abspath(
-            os.path.join(install_path, "_external_src", _GRIDTOOLS_REPO_DIRNAMES)
-        )
-        if not os.path.exists(target_path):
+    GIT_BRANCH = _GRIDTOOLS_GIT_BRANCH
+    GIT_REPO = _GRIDTOOLS_GIT_REPO
+
+    install_path = os.path.dirname(__file__)
+    external_path = os.path.abspath(os.path.join(install_path, "_external_src"))
+    target_path = os.path.join(external_path, _GRIDTOOLS_REPO_DIRNAMES)
```

Comment:
There is no guarantee that `gt_config.GT_REPO_DIRNAME` is a parent directory of `gt_config.build_settings["gt_include_path"]`. In other words, this logic will give unintended behavior depending on how gt_config is configured. The sources could very well get cloned, but still print saying they're not installed - I'm not sure whether they would be accessible for compilation in that case.

Reply:
You're right that we expose a lever here that when set gives some unintended behavior. The value of the `gt_include_path` key in the `build_settings` dict is:
```
os.environ.get("GT_INCLUDE_PATH", GT_INCLUDE_PATH)
```
so the intended usage is that the user sets the env var `GT_INCLUDE_PATH` instead of adding a new key to the global state dict.

Is that what you are getting at? I might be misunderstanding the point.

Reply:
> If I set GT_INCLUDE_PATH to anything other than its default, this function will still be called and will still clone sources and report on whether it was successful

Are you positive this always happens? I'm only able to reproduce when I set `GT_INCLUDE_PATH` to one that does not contain `gridtools/common/defs.hpp`.